### ISSUE TITLE
A process and policy for allowing official mirroring of releases

### ISSUE THREAD

**Issue by ftyers** [common-voice/common-voice#3262]
**Is your feature request related to a problem? Please describe.**

*Note: This is a placeholder issue, I don't expect anything actionable in the foreseeable future, but I wanted to get my thoughts down as it is something I have thought about for quite some time.*

The current system that requires people to get the data directly from Mozilla and provide an email address can be quite frustrating for people, especially those on low bandwidth or low reliability connections. See for example [this thread](https://discourse.mozilla.org/t/download-common-voice-corpus-7-0-en-2637h-2021-07-21-65gb-fails/85879) on Discourse. There are many others.

The motivation for not allowing people to mirror the content is (to my knowledge) based on two factors:

- If there are many mirrors it is difficult to maintain download statistics which makes it difficult to maintain funding for Common Voice, e.g. "millions of downloads/year" looks very good to funders.
- There are privacy concerns around having a way to contact people who have downloaded the dataset to get them to remove data from a specific person.

Given the current licence there is no way to legally or technically prevent people from mirroring the data (bypassing the email check is trivial), but most people who would want to mirror it are supporters of Common Voice and wouldn't want to do it without permission / understanding.

**Describe the solution you'd like**

The ideal thing would be for people to be able to mirror the content with the blessing of the CV team. Mirrors could agree to provide download statistics and sign a release saying that they will collect as much information as necessary to be able to contact downloaders with takedown requests.

Mirroring data is not an unusual situation and is very often done on a voluntary basis. In addition, it will make it easier for people to download, and give them faster downloads.

Note that without an official solution, people will just take the CC-0 licence at its value and mirror copies of the data anyway, and then control will be completely lost.

**Comment by DewiBrynJones** [comment]
I agree there needs to be an official solution to allow for mirroring.

Just to add to related thread and issues:

Downloading via the HuggingFace Dataset library is very convenient. 

https://huggingface.co/datasets/common_voice

But it's stuck at present on 6.1. GitHub issue https://github.com/huggingface/datasets/issues/2730

**Comment by ftyers** [comment]
It seems that HuggingFace is not following the Mozilla requirements to collect information about people downloading and in addition does not provide licensing information. It is nice that they give the citation though.

**Comment by patrickvonplaten** [comment]
We would be happy to try helping in any way that would allow easier access to the Common Voice 7 dataset at Hugging Face.

We're always trying to add licencing to the datasets: https://huggingface.co/datasets/common_voice and got the direct download links from Facebook's CoVost repo: https://github.com/facebookresearch/covost/blob/1e9b5269733c7f62eb784239c215acadc63bf8fd/README.md#covost-2

I understand that statistics about the data downloads can be important for funding and I think we would be happy to discuss a suitable integration with `datasets`  at Hugging Face if there is interest

**Comment by robovoice1** [comment]
Other ways of downloading via p2p - torrent and so on.
The hash files make sure you are downloading the official corpus release from moz/cv.
Make those official corpus data sets password protected. After downloading and trying to unpack it you have to type in a long password. To get this password you have to register on moz/cv website.
P2p would be interesting because power downloaders are kicked from their contracts.

> you have to register

**Comment by farooqkz** [comment]
Continuing discussion from #5098.

The people have the right to be forgotten. But all of us know that once the dataset is released, there is no way to prevent people from using some clips. All you can do is ask them to remove those clips. But in the end, it's up to people. Also you cannot prevent people from re-sharing the datasets. In fact, there are already examples in which this is being done.

The point I have is that telling people to not mirror these, will hugely limit the utility and benefit of the datasets. And the benefits doing so has, IMO, is very little.

For instance, I can do some processing on the dataset and re-share the modified version. But as the terms say I can't reshare, I'll have to publish the processing code, and everyone will have to do this themselves. Actually I think MLcommons is already doing so for multi lingual spoken words:

https://mlcommons.org/datasets/multilingual-spoken-words/

**Comment by ftyers** [comment]
@farooqkz thanks for this, derivative works is a good point. You are welcome to create derivative works and also host them on MDC (when upload functionality is implemented, which should be soon).

**Comment by ftyers** [comment]
Upload functionality is now implemented on [MDC](https://datacollective.mozillafoundation.org/).  There is now [programmatic access to datasets](https://github.com/Mozilla-Data-Collective/datacollective-python) with resumable downloads.
